src/main/java/com/htisolutions/poolref/controllers/ResetPasswordController.java

```diff
@@ -0,0 +1,57 @@
+package com.htisolutions.poolref.controllers;
+
+import com.htisolutions.poolref.services.ResetPasswordService;
+import com.htisolutions.poolref.viewModels.ResetPasswordNicknameViewModel;
+import com.htisolutions.poolref.viewModels.ResetPasswordQuestionViewModel;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.*;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.ModelAndView;
+
+@Controller
+@RequestMapping("/reset-password")
+public class ResetPasswordController {
+
+    private ResetPasswordService resetPasswordService;
+    private String nickname;
+
+    @Autowired
+    ResetPasswordController(ResetPasswordService resetPasswordService){this.resetPasswordService = resetPasswordService;}
```

Comment:
The methods in this controller are very confusing, you should just have two?

index()
reset()